pymc3/distributions/logprob.py

```diff
@@ -141,8 +129,89 @@ def logpt(
         A scaling term to apply to the generated log-likelihood graph.
     transformed
         Apply transforms.
-    cdf
-        Return the log cumulative distribution.
+
+    """
+    if not isinstance(rv_values, Mapping):
+        rv_values = (
+            {var: at.as_tensor_variable(rv_values).astype(var.type)}
+            if rv_values is not None
+            else {}
+        )
+
+    tmp_rv_values = {}
+    for node in io_toposort(graph_inputs((var,)), (var,)):
+        if isinstance(node.op, RandomVariable):
+            tmp_rv_values[node.out] = node.out
```

Comment:
This was required since joint_logprob` will simply ignore the RandomVariables which are not in the gen `rvs_to_values` list. Plus the transforms are not fully supported hence we still have to do the replacement logic below. (L-160 to L-167)